lib/parser_combinator/parserCombinator.ml

```diff
@@ -0,0 +1,52 @@
+open! ParserM
+
+module type ParserCombinatorSig = sig 
+  include ParserMSig
+  val (++) : 'a parser_m -> 'a parser_m -> 'a parser_m
+  val (>>) : 'a parser_m -> 'a parser_m -> 'a parser_m
+  val (<<) : 'a parser_m -> 'a parser_m -> 'a parser_m
+  val zero : error:string -> 'a parser_m
+  val item : token parser_m
+  val star : 'a parser_m -> 'a list parser_m
+  val plus : 'a parser_m -> 'a list parser_m
+  val sat : (token -> bool) -> ?error:string -> token parser_m
+end
+
+module Make (Monad : ParserMSig) : ParserCombinatorSig with type token = Monad.token = 
+struct
+  include Monad
+
+  let (++) p q = fun cs -> 
+    match p cs with 
+    | Ok _ as x -> x 
+    | Error _ -> q cs
+
+  let (>>) p q = 
+    let* _ = p in q 
```

Comment:
One idiom when using monads is if you are using `let*` (or normally, `let%bind`), to only use `let*` and finish with a `return`.

So here we would have
```
let* _ = p in
let* x = q in
return x
```

you can rewrite e.g. `(<<)` as well